### PR TITLE
Viite 2032 form of node point template

### PR DESCRIPTION
**Pull request by andre-policarpo**

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/NodePointDAO.scala [hunk 1]
```diff
@@ -192,6 +192,20 @@ class NodePointDAO extends BaseDAO {
...
+  def fetchNodePointTemplateById(id: Long): Option[NodePoint] = {
+    val query =
+      s"""
+         SELECT DISTINCT NP.ID, NP.BEFORE_AFTER, NP.ROADWAY_POINT_ID, NULL AS NODE_ID, NP.START_DATE, NP.END_DATE,
+         NP.VALID_FROM, NP.VALID_TO, NP.CREATED_BY, NP.CREATED_TIME, RP.ROADWAY_NUMBER, RP.ADDR_M, RW.ROAD_NUMBER, RW.ROAD_PART_NUMBER, RW.TRACK, rw.ELY
+         FROM NODE_POINT NP
+         JOIN ROADWAY_POINT RP ON (RP.ID = ROADWAY_POINT_ID)
+         JOIN LINEAR_LOCATION LL ON (LL.ROADWAY_NUMBER = RP.ROADWAY_NUMBER AND LL.VALID_TO IS NULL)
+         LEFT JOIN ROADWAY RW ON (RP.ROADWAY_NUMBER = RW.ROADWAY_NUMBER)
+         where NP.id = $id
```
[the remark this answers is not in the document]
Comment (andre-policarpo) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/NodePointDAO.scala]: On it.

**Comment by andre-policarpo** [comment]
We should merge junction template form - onto this one. And test UI, and review the code to avoid duplicate code. And use the same _CSS_ for titles and labels.

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/NodePointDAO.scala [hunk 2]
```diff
@@ -192,6 +192,20 @@ class NodePointDAO extends BaseDAO {
...
+  def fetchNodePointTemplateById(id: Long): Option[NodePoint] = {
+    val query =
+      s"""
+         SELECT DISTINCT NP.ID, NP.BEFORE_AFTER, NP.ROADWAY_POINT_ID, NP.NODE_ID, NP.START_DATE, NP.END_DATE,
+         NP.VALID_FROM, NP.VALID_TO, NP.CREATED_BY, NP.CREATED_TIME, RP.ROADWAY_NUMBER, RP.ADDR_M, RW.ROAD_NUMBER, RW.ROAD_PART_NUMBER, RW.TRACK, rw.ELY
+         FROM NODE_POINT NP
+         JOIN ROADWAY_POINT RP ON (RP.ID = ROADWAY_POINT_ID)
+         JOIN LINEAR_LOCATION LL ON (LL.ROADWAY_NUMBER = RP.ROADWAY_NUMBER AND LL.VALID_TO IS NULL)
+         LEFT JOIN ROADWAY RW ON (RP.ROADWAY_NUMBER = RW.ROADWAY_NUMBER)
+         where NP.id = $id AND NP.node_id is null AND NP.node_id is null and NP.valid_to is null and NP.end_date is null and NP.node_id is null and RW.end_date is null
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/NodePointDAO.scala]: One `NP.node_id is null` is enough.

**Comment by Campanha** [comment]
Form of node point template is not opening after selection in node point template list.

**Comment by Campanha** [comment]
event **nodeSearchTool:clickNodePointTemplate** seems to be triggered always 10 times consecutively.

**Comment by Campanha** [comment]
looks good now

**Pull request closed without merging**